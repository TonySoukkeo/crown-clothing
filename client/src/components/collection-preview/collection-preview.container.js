import { withRouter } from "react-router-dom";

import CollectionPreview from "./collection-preview.component";

const CollectionPreviewContainer = withRouter(CollectionPreview);

export default CollectionPreviewContainer;
