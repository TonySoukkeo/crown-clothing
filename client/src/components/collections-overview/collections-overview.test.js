import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import CollectionsOverview from "./collections-overview.component";
import { default as CollectionPreview } from "../collection-preview/collection-preview.container";

configure({
  adapter: new Adapter(),
});

describe("CollectionsOverview component", () => {
  let wrapper;
  let mockCollections;

  beforeEach(() => {
    mockCollections = [
      {
        id: 1,
      },
      { id: 2 },
      { id: 3 },
    ];

    wrapper = shallow(<CollectionsOverview collections={mockCollections} />);
  });

  it("renders CollectionsOverview without crashing", () => {
    expect.assertions(1);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders out correct number of CollectionPreview elements", () => {
    const collectionPreview = wrapper.find(CollectionPreview);

    expect.assertions(1);
    expect(collectionPreview.length).toEqual(mockCollections.length);
  });
});
