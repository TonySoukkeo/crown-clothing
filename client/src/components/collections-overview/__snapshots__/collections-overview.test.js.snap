// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview component renders CollectionsOverview without crashing 1`] = `ShallowWrapper {}`;
